src/viewer.js

```diff
@@ -126,21 +136,23 @@ async function generateReport(bundleStats, opts) {
     openBrowser = true,
     reportFilename,
     reportTitle,
+    compressionAlgorithm = 'gzip',
```

Comment:
Oh yeah, I see you've found that we have multiple places where we're defining default values for options, and haven't really made up our mind where they'd belong 😅 

How do you feel about moving all the `compressionAlgorithm = 'gzip` default value definitions to the top-level usages? That is, moving the `compressionAlgorithm = 'gzip'` code to the plugin starting point in `src/BundleAnalyzerPlugin.js`. The CLI starting point in `src/bin/analyzer.js` already seems to define the default value for the `--compression-algorithm` to be `gzip` so we should be covered there.

Then we could get rid of all the `compressionAlgorithm = 'gzip'` defaulting in other code paths and assume that we have the value filled in in those code paths.